### PR TITLE
issue-2513: Progress Indicator: fixed optionalText styling to make it consistent with react carbon component

### PR DESCRIPTION
**Pull request by rajeshkumaru**
Closes carbon-design-system/carbon-components-angular#2513

#### Changelog

**Before**

Screenshot 2023-04-07 at 2 07 40 PM

**Changed**

Vertical progress indicator:

Screenshot 2023-04-07 at 2 07 59 PM

Horizontal progress indicator:
Screenshot 2023-04-07 at 2 53 07 PM

### CODE REVIEW

**Comment by rajeshkumaru** [comment]
I have read the DCO document and I hereby sign the DCO.

File: src/progress-indicator/progress-indicator.component.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ import { Step } from "./progress-indicator-step.interface";
 			class="bx--progress-step bx--progress-step--{{step.state[0]}}"
 			*ngFor="let step of steps; let i = index"
 			[ngClass]="{'bx--progress-step--disabled' : step.disabled}">
-			<div class="bx--progress-step-button bx--progress-step-button--unclickable" role="button" tabindex="-1">
+			<button class="bx--progress-step-button bx--progress-step-button--unclickable" role="button" tabindex="-1">
```
Comment (rajeshkumaru) [inline src/progress-indicator/progress-indicator.component.ts:31]: Converted to `button` to fix gap btw text and optionalText. (react has button)
Screenshot 2023-04-07 at 2 51 54 PM

File: src/progress-indicator/progress-indicator.component.ts [hunk 2]
```diff
@@ -48,10 +48,12 @@ import { Step } from "./progress-indicator-step.interface";
 					(click)="stepSelected.emit({ step: step, index: i })">
 					{{step.text}}
 				</p>
-				<p class="bx--progress-label" *ngIf="!step.tooltip" (click)="stepSelected.emit({ step: step, index: i })">{{step.text}}</p>
-				<p *ngIf="step.optionalText" class="bx--progress-optional">{{step.optionalText}}</p>
+				<div class="bx--progress-text">
```
Comment (Akshat55) [inline src/progress-indicator/progress-indicator.component.ts:51]:
```suggestion
				<div class="bx--progress-text" *ngIf="!step.tooltip || step.optionalText">
```

If both are false, we shouldn't render an empty node.

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/progress-indicator/progress-indicator.component.ts (hunk 2 above)
Comment (rajeshkumaru) [reply src/progress-indicator/progress-indicator.component.ts]: @Akshat55 , i think we might need this condition instead.

```
*ngIf="(!step.tooltip && step.optionalText) || step.optionalText"
```

Comment (rajeshkumaru) [reply src/progress-indicator/progress-indicator.component.ts:51]: hmm..i don't think there is a cleaner way to avoid rendering empty node.

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**